GH-622 Fix Windows pathing issues + clean up (#127)
eclipse-glsp/glsp-examples: project-templates/java-emf-theia/README.md

-   [`glsp-client/tasklist-browser-app/package.json`](glsp-client/tasklist-browser-app/package.json): Theia browser application definition
-   [`glsp-server/src/main/java/org/eclipse/glsp/example/javaemf/TaskListDiagramModule.java`](glsp-server/src/main/java/org/eclipse/glsp/example/javaemf/TaskListDiagramModule.java): dependency injection module of the server

> > **_NOTE:_**&nbsp; Due to bug [GLSP-666](https://github.com/eclipse-glsp/glsp/issues/666) the launch configurations for the `Theia Backend` might not work as expected when using Windows. Unfortunately there is currently no work-around and if you encounter this bug you won't be able to debug the Theia backed.

Reviewer: ```suggestion
> **_NOTE:_**&nbsp; Due to bug [GLSP-666](https://github.com/eclipse-glsp/glsp/issues/666) the launch configurations for the `Theia Backend` might not work as expected on Windows.If you encounter this bug, the Theia backend will start and function as expected, but you won't be able to debug the Theia backend on Windows.
```

Author: That's not true. Currently the backend won't work start and function as expected. For this we would have to enable the `PRESERVE_SYMLINKS` option for Windows. However, if we do this we would have to rework the way we receive the path for the server executable. Currently we do this with `path.join(__dirname, '../../../../glsp-server/lib/index.js'),`
However `__dirname` is different depending on whether `PRESERVE_SYMLINKS` is enabled or not.

Reviewer: Ah, ok sorry, I wasn't aware of that. Then please disregard my comments regarding this sentence.

## Building the example

> The server component hast to be built using `maven` and the client component have to be built using `yarn`.

Reviewer: ```suggestion
The server component is built with `maven` and the client component is built with `yarn`.
```
